src/printer.js

```diff
@@ -764,16 +764,29 @@ function genericPrintNoParens(path, options, print) {
         const needsForcedTrailingComma = canHaveTrailingComma &&
           lastElem === null;
 
+        var printedElements = [];
+        let separatorParts = [];
+        path.each(
+          function(childPath) {
+            printedElements.push(concat(separatorParts));
+            printedElements.push(group(print(childPath)));
+
+            separatorParts = [",", line];
+            if (childPath.getValue() &&
+                util.isNextLineEmpty(options.originalText, childPath.getValue())) {
+              separatorParts.push(hardline);
```

Comment:
This also effectively forces arrays to be expanded if they were expanded in the original source.